catalog/glue/glue.go

```diff
@@ -155,33 +156,42 @@ func NewCatalog(opts ...Option) *Catalog {
 // ListTables returns a list of Iceberg tables in the given Glue database.
 //
 // The namespace should just contain the Glue database name.
-func (c *Catalog) ListTables(ctx context.Context, namespace table.Identifier) ([]table.Identifier, error) {
+
+func (c *Catalog) ListTables(ctx context.Context, namespace table.Identifier) iter.Seq2[table.Identifier, error] {
+	return func(yield func(table.Identifier, error) bool) {
+		for {
+			tbls, nextPageToken, err := c.ListTablesPage(ctx, namespace)
+			if err != nil {
+				yield(table.Identifier{}, err)
+				return
+			}
+			for _, tbl := range tbls {
+				if !yield(tbl, nil) {
+					return
+				}
+			}
+			if nextPageToken == nil {
+				return
+			}
+		}
+	}
+}
+
+func (c *Catalog) ListTablesPage(ctx context.Context, namespace table.Identifier) ([]table.Identifier, *string, error) {
```

Comment:
Any particular reason to export this?

Also I'm not seeing how the next page token is being passed back to it for it to be used? 

In addition, any reason to use a pointer to the string rather than just using an empty string to indicate there are no further pages?

Reply:
Thanks for the quick review. I really appreciate it. :pray: 
>Any particular reason to export this?

No just typo on my side. It should be private method
>Also I'm not seeing how the next page token is being passed back to it for it to be used?

Cmiiw, from the current code, the `c.glueSvc.GetTables` does not use any next page token. Mainly I think it is meant for checking the end of the page in `ListTables`